Client/API/ApiRequestHandler.cs

```diff
@@ -52,11 +52,11 @@ public async Task<IEnumerable<Country>> FetchAllCountries(HttpClient httpClient)
             }
         }
 
-        public async Task<Country> FetchCountryByYear(HttpClient httpClient, string code, DateOnly year)
+        public async Task<Country> FetchCountryByYear(HttpClient httpClient, string code, DateOnly date)
         {
             try
             {
-                return await httpClient.GetFromJsonAsync<Country>($"{apiUrl}/Country/GetCountryDataForYear/{code}/{year}");
+                return await httpClient.GetFromJsonAsync<Country>($"{apiUrl}/Country/GetCountryDataForYear/{code}/{date.Year}-{date.Month}-{date.Day}");
```

Comment:
the only data related to date that we recieve from our data is "Year", there are no month or days included.